tests/integration/test_network_configuration.py

```diff
@@ -0,0 +1,248 @@
+"""
+This test file captures the _current_ state of returning URLs before making
+sweeping changes. This is to ensure that the refactoring does not cause
+external breaking behaviour. In the future we can update this test suite to
+correspond to the behaviour we want, and we get a todo list of things to
+change 😂
+"""
+import json
+
+import pytest
+import requests
+import xmltodict
+from botocore.auth import SigV4Auth
+
+from localstack import config
+from localstack.aws.api.lambda_ import Runtime
+from localstack.testing.aws.lambda_utils import is_new_provider, is_old_provider
+from localstack.utils.files import new_tmp_file, save_file
+from localstack.utils.strings import short_uid
+
+# TODO: how do we test `localstack_hostname` - this variable configures the
+# host that services make requests to when starting up (e.g. opensearch) and
+# they won't start if we override the variable.
```

Comment:
This seems to be resolved with the usage of `socket.gethostname` in the `fixtures.py`.